ckanext/canada/plugin/public_plugin.py

```diff
@@ -212,6 +215,56 @@ def column_types(self, existing_types: Dict[str, Type[ColumnType]]):
     def get_blueprint(self) -> List[Blueprint]:
         return [canada_views]
 
+    def _canada_citation_map(self, cite_data: DataDict,
+                             pkg_dict: DataDict) -> Dict[str, Any]:
+        lang = 'en'
+        if has_request_context():
+            lang = h.lang()
+
+        if (
+          pkg_dict.get('org_section') and
+          lang in pkg_dict['org_section']
+          and pkg_dict['org_section'][lang]):
+            cite_data['publisher'] += ' - ' + pkg_dict['org_section'][lang]
+
+        cite_data['author'] = []
+        if pkg_dict.get('creator'):
+            cite_data['author'].append({
+                'family': pkg_dict['creator']
+            })
+        if (
+          pkg_dict.get('contributor') and
+          lang in pkg_dict['contributor'] and
+          pkg_dict['contributor'][lang]):
+            cite_data['author'].append({
+                'family': pkg_dict['contributor'][lang]
+            })
+        if pkg_dict.get('credit'):
+            for e in pkg_dict['credit']:
+                if lang not in e['credit_name'] or not e['credit_name'][lang]:
+                    continue
+                cite_data['author'].append({
+                    'family': e['credit_name'][lang]
+                })
+        if not cite_data['author']:
+            cite_data.pop('author', None)
+
+        return cite_data
+
+    # ICiteProcMappings
+    def dataset_citation_map(self, cite_data: DataDict,
+                             pkg_dict: DataDict) -> Tuple[bool, Dict[str, Any]]:
+        cite_data['container_title'] = _(cite_data['container_title'])
+        cite_data = self._canada_citation_map(cite_data, pkg_dict)
+        return False, cite_data
```

Comment:
modifying a passed parameter and *also* returning that parameter is an anti-pattern. I know a number of existing interfaces in ckan do this but we shouldn't repeat this for new interfaces.